src/hotspot/share/code/aotCodeCache.cpp

```diff
@@ -4592,18 +4592,18 @@ address SCAddressTable::address_for_id(int idx) {
   if (id >= _c_str_base && id < (_c_str_base + (uint)_C_strings_count)) {
     return address_for_C_string(id - _c_str_base);
   }
-  fatal("Incorrect id %d for SCA table", id);
+  fatal("Incorrect id %d for AOT Code Cache addresses table", id);
   return nullptr;
 }
 
-int SCAddressTable::id_for_address(address addr, RelocIterator reloc, CodeBuffer* buffer) {
+int AOTCodeAddressTable::id_for_address(address addr, RelocIterator reloc, CodeBuffer* buffer) {
+  if (!_extrs_complete) {
+    fatal("AOT Code Cache VM runtime addresses table is not complete");
+  }
```

Comment:
We sure this move makes sense?

Reply:
`id_for_address()` is called for processing relocation info in compiled code (stubs, adapters, code). We should not do that until we initialized the table for VM's runtime addresses at least! We do that initialization very early in `SCCache::init2()` which is called just after `universe_init()`.

Reply:
OK!

Reply:
I looked on my original changes and I think I placed the check there because `int id = -1;` was first instruction.
But I will run more testing to see when `-1` is passed to make sure this is fine.